libs/designsystem/src/lib/components/button/button.component.ts

```diff
@@ -71,18 +70,27 @@ export class ButtonComponent implements AfterContentInit {
 
   ngAfterContentInit(): void {
     if (this.iconDomNode && this.iconDomNode.nativeElement) {
-      const prev = this.iconDomNode.nativeElement.previousSibling;
-      const next = this.iconDomNode.nativeElement.nextSibling;
-      if (prev && prev.nodeName !== '#comment') {
+      const iconNativeElement = this.iconDomNode.nativeElement;
+      const prev = this.findNonCommentSibling(iconNativeElement, 'previousSibling');
+      const next = this.findNonCommentSibling(iconNativeElement, 'nextSibling');
+      if (prev) {
         this._hasSlottedContent = true;
         this._isIconLeft = false;
         this._isIconRight = true;
       }
-      if (next && next.nodeName !== '#comment') {
+      if (next) {
         this._hasSlottedContent = true;
         this._isIconLeft = true;
         this._isIconRight = false;
       }
     }
   }
+
+  private findNonCommentSibling(nativeElement: any, direction: 'nextSibling' | 'previousSibling') {
```

Comment:
You can strong type the `nativeElement` parameter by typing the `iconDomNode` member var as:
 `iconDomNode: ElementRef<HTMLElement>;`

Reply:
Done! - however had to cast the `sibling` variable in `findNonCommentSibling` to a `HTMLElement` for it to not complain. I personally don't think this is an issue in this context, as it is happening inside the function, and is "hidden" from the consumer. 

But wouldn't mind your toughts on it 😄

Reply:
note: typecast has been moved as part of the argument when doing the recursive call to `findNonCommentSibling`.